samples/videoDecodeBatch/videodecodebatch.cpp

```diff
@@ -125,9 +125,11 @@ void DecProc(RocVideoDecoder *p_dec, VideoDemuxer *demuxer, int *pn_frame, doubl
         n_frame_returned = p_dec->DecodeFrame(p_video, n_video_bytes, 0, pts);
         n_frame += n_frame_returned;
         if (b_dump_output_frames && mem_type != OUT_SURFACE_MEM_NOT_MAPPED) {
-            if (!n_frame && !p_dec->GetOutputSurfaceInfo(&surf_info)) {
-                std::cerr << "Error: Failed to get Output Surface Info!" << std::endl;
-                break;
+            if(n_frame){
+                if(!p_dec->GetOutputSurfaceInfo(&surf_info)){
```

Comment:
The !n_frame was added since we wanted call GetOutputSurfaceInfo() only once for the first frame where n_frame will be 0. The new code will call it for every frame. Please check this as well

Reply:
During testing on Pavel's system, it was found that GetOutputSurfaceInfo() sometimes never gets called and this results in random values (mostly 0) in surf_info and in turn results in 0 output file size (or huge size in rare case).

I think calling GetOutputSurfaceInfo() for every decode call is needed to deal with video size change with the stream, although not efficient.

Reply:
As for why GetOutputSurfaceInfo() never gets called, I think it can be from compiler optimization when n_frame != 0, !n_frame will be false and the call to GetOutputSurfaceInfo() is skipped. @paveltc Can you confirm on the stream that produces the issue by logging out n_frame as well?

Reply:
@jeffqjiangNew I will add the log and see what results I get.

Reply:
@jeffqjiangNew yes this a compiler optimization. I have seen it happen before. If the first condition in an AND case evaluates to false, it doesn't evaluate the second at all.

Reply:
@jeffqjiangNew I logged the amount of frames processed with and without my fix. The amount of frames given by the debug messages are the same for both. However, without my fix, the output of three of the videos is 0 bytes.

Reply:
@paveltc My question was to check if n_frame is non-zero for the first decode call, in order to confirm that GetOutputSurfaceInfo() is NOT called. So what is the value of n_frame for the first decode call on the stream with issue?

And do conformance tests, both AVC/HEVC tests pass?

Reply:
@jeffqjiangNew The problem streams give 1 for n_frame for the first call.

Reply:
@paveltc Thanks. This explains why GetOutputSurfaceInfo() is not called.